apps/main/src/modules/wallet/MyAssets/MyAssetsTable.tsx

```diff
@@ -1,68 +1,100 @@
+  MyAssetsTableColumn,
   useMyAssetsColumns,
 } from "@/modules/wallet/MyAssets/MyAssetsTable.columns"
 import { useAssets } from "@/providers/assetsProvider"
+import { useAccountData } from "@/states/account"
+import { scaleHuman } from "@/utils/formatting"
 
 type Props = {
   readonly searchPhrase: string
+  readonly showAllAssets: boolean
 }
 
-export const MyAssetsTable: FC<Props> = ({ searchPhrase }) => {
-  const { isMobile } = useBreakpoints()
-  const { tokens, stableswap, bonds, erc20, native, isExternal } = useAssets()
-  const columns = useMyAssetsColumns()
+export const MyAssetsTable = forwardRef(
+  ({ searchPhrase, showAllAssets }: Props, ref: ForwardedRef<DataTableRef>) => {
+    const { isMobile } = useBreakpoints()
 
-  const filteredTokens = useMemo(
-    () =>
-      [...tokens, ...stableswap, ...bonds, ...erc20].filter(
-        (asset) => asset.isTradable && !isExternal(asset),
-      ),
-    [tokens, stableswap, bonds, erc20, isExternal],
-  )
+    const { native, all, isExternal } = useAssets()
+    const balances = useAccountData((data) => data.balances)
 
-  // TODO integrate
-  const assets = useMemo(() => {
-    return filteredTokens.map<MyAsset>((asset) => ({
-      ...asset,
-      total: 12345678,
-      transferable: 12345678,
-      canStake: asset.symbol === "USDC",
-    }))
-  }, [filteredTokens])
+    const tableAssets = useMemo(() => {
+      const assetsWithBalance = showAllAssets
+        ? Array.from(all.values()).map((asset) => ({
+            ...asset,
+            balance: balances[asset.id],
+          }))
+        : Object.entries(balances)
+            .map(([assetId, balance]) => {
+              const asset = all.get(assetId)
 
-  return (
-    <TableContainer as={Paper}>
-      <DataTable
-        paginated
-        pageSize={10}
-        globalFilter={searchPhrase}
-        data={assets}
-        columns={columns}
-        expandable={!isMobile}
-        renderSubComponent={(asset) =>
-          asset.id === native.id ? (
-            <ExpandedNativeRow assetId={asset.id} />
-          ) : (
-            <AssetDetailExpanded assetId={asset.id} />
-          )
-        }
-        renderOverride={(asset) =>
-          // TODO rug check
-          asset.id === "10" ? <InvalidAssetRow assetId={asset.id} /> : undefined
-        }
-      />
-    </TableContainer>
-  )
-}
+              if (!asset) {
+                return null
+              }
+
+              return {
+                ...asset,
+                balance,
+              }
+            })
+            .filter((asset) => !!asset)
+
+      return assetsWithBalance
+        .filter((asset) => !isExternal(asset) || !!asset.name)
+        .map<MyAsset>((asset) => ({
+          ...asset,
+          total: scaleHuman(balances[asset.id]?.total ?? 0n, asset.decimals),
+          transferable: scaleHuman(
+            balances[asset.id]?.free ?? 0n,
+            asset.decimals,
+          ),
+          canStake: asset.symbol === "HDX",
```

Comment:
Can you compare here with asset.id and native.id. Or even better to have config which assets are stakable. But for now native.id should o, since I'm not sure we will have something else stakable